### ISSUE TITLE
[Bug][Azure devops] Can't add repositories

### ISSUE THREAD

**Issue by klemen-df** [apache/devlake#6401]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

- Create and successfully test a new connection to Azure Devops
- Add scope config
- - list of Projects is populated
- - when selecting a Project 'the wheel' keeps spining and repositories are not loaded

It does this for some of the projects but not for others.

Nothing in F12, only in logs :
`2023-11-06T14:03:47+01:00 time="2023-11-06 13:03:47" level=error msg="HTTP 500 error\n\tcaused by: unexpected end of JSON input\n\tWraps: (2) unexpected end of JSON input\n\tError types: (1) *hintdetail.withDetail (2) *json.SyntaxError"`

### What do you expect to happen

Repositories should be loaded.

### How to reproduce

- Create and successfully test a new connection to Azure Devops
- Add scope config
- - list of Projects is populated
- - when selecting a Project 'the wheel' keeps spining and repositories are not loaded

### Anything else

every time for the same project

### Version

v0.19.0-beta6

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Startrekzky** [comment]
Hi @keon94 , can you have a look?

**Comment by keon94** [comment]
@KlemenDanfoss 
Do these failing projects have a huge number of repos under them? Possibly over 200? This might be a related discussion and I have suggested a troubleshooting script there:
https://github.com/apache/incubator-devlake/issues/6244#issuecomment-1769877928

**Comment by gorhan99** [comment]
I have the same problem

**Comment by colinfrei** [comment]
I have the same issue as well, though it works for some Azure Devops Projects, and not for others (within the same organization).

**Comment by KlemenDanfoss** [comment]
> @KlemenDanfoss Do these failing projects have a huge number of repos under them? Possibly over 200? This might be a related discussion and I have suggested a troubleshooting script there: [#6244 (comment)](https://github.com/apache/incubator-devlake/issues/6244#issuecomment-1769877928)

104 repos.

**Comment by keon94** [comment]
I've recreated this problem by generating a dummy list of about 500 repos. Basically it boils down to a huge payload size being sent from Python to Go, which is either extremely slow or silently fails somewhere. This is pretty simple to solve with the stream API I've created in the past for Go<->Python. This way repos will be sent one by one (tested and works in just a few seconds). A better solution is to stream groups of repos in batches for better efficiency which I am looking into right now.
